web/style/common.scss

```diff
@@ -0,0 +1,284 @@
+    height: $header-height;  
+  }
+  @include break-point(desktop) {
+    height: $header-height;  
+  }
+  box-shadow: 0 8px 6px -3px $grey-color-3;
+  @include app-header-padding-changes;
+
+  &__logo {
+    margin-left: 1vw;
+    width: 20vw;
+    padding: 9.5px 0;
+    @include break-point(tablet) {
+      margin-left: 0;
+      width: 23vw;
+    }
+
+    @include break-point(desktop) {
+      margin-left: 0;
+      width: 13vw;
+    }
+  }
+  &__bar {
+    @include flex-row;
+    list-style: none;
+    margin-left: 7vw;
+    padding: 9.5px 0;
+
+    &--link {
+      @include flex-column-reverse;
+      margin-left: 2vw;
+      display: none;
+      @include break-point(tablet) {
+        display: flex;
+      }
+      a {
+        color: #736a6d;
+      }
+    }
+  }
+
+  &__end {
+    @include flex;
+    flex-direction: column;
+    width: 29vw;
+    @include break-point(tablet) {
+      width: 12vw;
+    }
+
+    &-register-box {
+      @include flex;
+      // flex-direction: row-reverse;
+      height: 34%;
+      display: none;
+      @include break-point(tablet) {
+        display: flex;
+      }
+
+      &--link {
+        font-size: 0.7rem;
+        margin-left: 1.5vw;
+      }
+    }
+    &-cart {
+      @include flex-row;
+      justify-content: center;
+      align-items: center;
+      background-color: $grey-color-3;
+
+      &--logo {
+        height: 0.37 * $header-height;
+        fill: $background-color-secondary;
+      }
+    }
+  }
+}
+
+// toggle menu for mobile
+.menu-toggle {
+  display: block;
+  position: relative;
+  z-index: 1;
+  -webkit-user-select: none;
+  user-select: none;
+  align-self: center;
+  @include break-point(tablet) {
+    display: none;
+  }
+
+  input {
+    display: block;
+    width: 40px;
+    height: 32px;
+    position: absolute;
+    top: -7px;
+    left: -5px;
+    cursor: pointer;
+    opacity: 0;
+    z-index: 2;
+    -webkit-touch-callout: none;
+    &:checked ~ span {
+      opacity: 1;
+      transform: rotate(45deg) translate(-2px, -1px);
+      background: #232323;
+    }
+    &:checked ~ span:nth-last-child(3) {
+      opacity: 0;
+      transform: rotate(0deg) scale(0.2, 0.2);
+    }
+    &:checked ~ span:nth-last-child(2) {
+      transform: rotate(-45deg) translate(0, -1px);
+    }
+    &:checked ~ ul {
+      display: block;
+      transform: translate(-150%, 0);
+    }
+  }
+  span {
+    display: block;
+    width: 33px;
+    height: 4px;
+    margin-bottom: 5px;
+    position: relative;
+    background: $grey-color;
+    border-radius: 3px;
+    z-index: 1;
+    transform-origin: 4px 0px;
+    transition: transform 0.5s cubic-bezier(0.77, 0.2, 0.05, 1),
+      background 0.5s cubic-bezier(0.77, 0.2, 0.05, 1), opacity 0.55s ease;
+    &:first-child {
+      transform-origin: 0% 0%;
+    }
+    &:nth-last-child(2) {
+      transform-origin: 0% 100%;
+    }
+  }
+}
+
+// mobile menu toggle list
+.menu {
+  $menu-width: 300px;
+
+  display: none;

```

Comment:
If Menu is display: none, why do you need so much of CSS for it? This CSS should be applied when menu is visible through and open/active class

Reply:
on checkbox clicked this css will work on checkox basis I mange display `block ` or `none`